fix(release-preflight): swift build --build-tests can't find Testing module under CLT-only

## What broke

\`scripts/release-preflight.sh\` step 6 fails with:

\`\`\`
Tests/EnviousWisprASRTests/LanguageDetectorTests.swift:3:8: error: no such module 'Testing'
FAIL: Test target failed to compile
\`\`\`

Encountered during the v1.9.4 release preflight on 2026-04-17.

## Why

We run CLT-only (no Xcode). Bare \`swift build --build-tests\` doesn't add \`Testing.framework\` or \`lib_TestingInterop.dylib\` to the search/link path, so Swift Testing-based tests fail to compile.

\`scripts/swift-test.sh\` already knows how to work around this — it adds the necessary \`-Xswiftc\` / \`-Xlinker -rpath\` flags pointing at the CLT Frameworks dir and Developer/usr/lib. That wrapper compiles AND runs all 334 tests cleanly (confirmed during v1.9.4 release).

The preflight script was last modified 2026-03-23 and predates the Testing framework adoption (or predates a CLT version bump that exposed this).

## Fix options

1. **Preferred:** change preflight step 6 from \`swift build --build-tests\` to \`scripts/swift-test.sh --no-run\` or equivalent — reuse the wrapper's flag logic.
2. Alternative: inline the same \`-Xswiftc/-Xlinker\` flags in the preflight script.

## Workaround for now

v1.9.4 shipped by running \`scripts/swift-test.sh\` manually (green, 334 tests) and verifying other preflight steps by hand. That is not a sustainable pattern.

## Scope

Non-blocking. SMALL tier, pure ops fix. One-line change to scripts/release-preflight.sh.

--- a/scripts/release-preflight.sh
+++ b/scripts/release-preflight.sh
@@ -60,9 +60,13 @@ if ! swift build -c release --arch arm64 2>&1 | tee /tmp/release-preflight-build
 fi
 echo "  OK  Release build passed"
 
-# 6. Test target compiles
-echo "  ... Running swift build --build-tests"
-if ! swift build --build-tests 2>&1 | tail -3; then
+# 6. Test target compiles.
+# Use scripts/swift-test.sh --no-run so the CLT Testing.framework flags are
+# applied. Bare `swift build --build-tests` fails under CLT-only because
+# Testing.framework and lib_TestingInterop.dylib aren't on the default path.
+echo "  ... Running scripts/swift-test.sh --no-run"
+if ! BUILD_OUTPUT=$(scripts/swift-test.sh --no-run 2>&1); then
+    echo "$BUILD_OUTPUT" | tail -10
     echo "FAIL: Test target failed to compile"
     exit 1
 fi

--- a/scripts/swift-test.sh
+++ b/scripts/swift-test.sh
@@ -16,6 +16,7 @@
 # Usage:
 #   scripts/swift-test.sh              # run all tests
 #   scripts/swift-test.sh --filter Foo # filter to tests matching "Foo"
+#   scripts/swift-test.sh --no-run     # compile tests only (used by release-preflight)
 
 set -euo pipefail
 
@@ -30,10 +31,26 @@ if [ ! -d "$FRAMEWORKS_DIR/Testing.framework" ]; then
     exit 1
 fi
 
-exec swift test \
-    -Xswiftc "-F${FRAMEWORKS_DIR}" \
-    -Xlinker "-F${FRAMEWORKS_DIR}" \
-    -Xlinker -rpath -Xlinker "${FRAMEWORKS_DIR}" \
-    -Xlinker -rpath -Xlinker "${USR_LIB_DIR}" \
-    -Xlinker -rpath -Xlinker "${INTEROP_LIB_DIR}" \
-    "$@"
+# Shared flag set — single source of truth for CLT Testing.framework wiring.
+CLT_FLAGS=(
+    -Xswiftc "-F${FRAMEWORKS_DIR}"
+    -Xlinker "-F${FRAMEWORKS_DIR}"
+    -Xlinker -rpath -Xlinker "${FRAMEWORKS_DIR}"
+    -Xlinker -rpath -Xlinker "${USR_LIB_DIR}"
+    -Xlinker -rpath -Xlinker "${INTEROP_LIB_DIR}"
+)
+
+NO_RUN=0
+REMAINING_ARGS=()
+for arg in "$@"; do
+    case "$arg" in
+        --no-run) NO_RUN=1 ;;
+        *)        REMAINING_ARGS+=("$arg") ;;
+    esac
+done
+
+if [[ $NO_RUN -eq 1 ]]; then
+    exec swift build --build-tests "${CLT_FLAGS[@]}" "${REMAINING_ARGS[@]}"
+fi
+
+exec swift test "${CLT_FLAGS[@]}" "${REMAINING_ARGS[@]}"

